### ISSUE TITLE
Opening the target editor loads the target platform

### ISSUE THREAD

**Issue by vogella** [eclipse-pde/eclipse.pde#616]
If I open the target editor to "quickly" check something, the re-loading of the target platform is triggered. I suggest we only do this on request, as this can be time consuming.

**Comment by mickaelistria** [comment]
Most of the editor functionality relies on a resolved target platform. However, some bits are already usable before the target is loaded. Can you please elaborate on what particular feature is not available early enough? Instead of disabling resolution, we can just fix the enablement of the particular feature.

**Comment by vogella** [comment]
The target platform is already resolved. Opening the editor re-triggers the resolving (which makes the target invalid during the resolve process).

**Comment by laeubi** [comment]
Didn't we already discussed this somewhere? I think the conclusion was that actually one wants to make resolution faster but not much progress on this right now :-)

But what you are talking about here is more that the **current active target state** is invalidated, what I think should not be the case but only happen when requested.

**Comment by merks** [comment]
It does sound quite similar to this issue:

https://github.com/eclipse-pde/eclipse.pde/issues/452

The only new thing here appears to be the comment about the target being invalid during the resolution.  How does this problem of the target being invalid during resolution manifest itself?

**Comment by laeubi** [comment]
I also noticed this some time it you can reproduce it with a target that uses a remote site (e.g. eclipse sdk), set it as a target and have a bundle require something from it.
Then keep the targeteditor open and close eclipse, make sure not internet connection is there and start eclipse again, it will try to resolve the target-editor and the workspace shows errors because of missing items as the target editor resolution fails.

**Comment by kysmith-csg** [comment]
> this can be time consuming.

It can be worse than this too. I was working on an airplane once with no internet. The target was fully resolved and everything was fine. I opened the editor to check the version of something (or similar) and it tried+failed to resolve, and made my entire workspace invalid.

I've also had it resolve I think sometimes from just switching from the source tab back to the visual editor.

Another idea would be that if something in one location is changed (say, I add a Maven dependency), don't resolve the _entire_ platform again, just the parts that were changed.

**Comment by merks** [comment]
Yes, I hate that.  It is possible to do it differently.    E.g., comparing to https://wiki.eclipse.org/Oomph_Targlets 

> The resolution process supports roll-back on failure.
> 
> - A resolution failure or network failure with Targlets will leave the old successful resolution in place.
> - A resolution failure with PDE's Software Site container will destroy your target platform, leaving all your workspace projects with errors until the problem is resolved, e.g., until your network or the hosting servers of the software sites are restored.

I'm generally quiet careful when I open a *.target file and first think, "do I need PDE to try to download all the files in the target in order to provide an outline?"  But that's a small consolation after the fact.  

In any case, I don't think it's easy to rework PDE's current implementation to use a p2 profile that records the information, that can display that information without doing anything other than loading the recorded information, and is only updated upon successful resolution.  Perhaps something simpler could be done, but the people to do that work are hard to come by...

**Comment by niport** [comment]
Actually, sometimes the target platform is reloaded (resolved) when I launch eclipse, **even if the target platform definition file is not opened in any editor**. In Oxygen this didn't happen.

If the machine is not connected or the target platform sites are not reachable (for instance because I am inside a company VPN), the resolution process fails. The issue here is that in any case PDE tries to build the project.

As a side effect PDE generates again the DS components and updates all the MANIFEST.MF files of the plugins that have at least a  DS component inside (the new MANFEST.MF files are different because Eclipse moves the Service-Component bundle header in a different position).

As a general rule if nothing is changed in the workspace, nothing have to be build or if something is build, since the sourcecode is not changed the derived file must not be changed.

I had already reported the side effects I described here in issue #74, but the cause is that the target platform is resolved on Eclipse startup and a full rebuild is started even if the target platform is not resolved at all!
